Examples/Algorithms/Geant4/src/Geant4Manager.cpp

```diff
@@ -0,0 +1,143 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Geant4/Geant4Manager.hpp"
+
+#include "ActsExamples/Geant4/PhysicsListFactory.hpp"
+
+#include <memory>
+#include <stdexcept>
+
+#include <FTFP_BERT.hh>
+#include <FTFP_BERT_ATL.hh>
+#include <G4EmParameters.hh>
+#include <G4HadronicParameters.hh>
+#include <G4HadronicProcessStore.hh>
+#include <G4Profiler.hh>
+#include <G4RunManager.hh>
+#include <G4RunManagerFactory.hh>
+#include <G4UserEventAction.hh>
+#include <G4UserRunAction.hh>
+#include <G4UserSteppingAction.hh>
+#include <G4UserTrackingAction.hh>
+#include <G4VUserDetectorConstruction.hh>
+#include <G4VUserPhysicsList.hh>
+#include <G4VUserPrimaryGeneratorAction.hh>
+#include <G4Version.hh>
+
+namespace ActsExamples {
+
+Geant4Handle::Geant4Handle(int _logLevel,
+                           std::unique_ptr<G4RunManager> _runManager,
+                           std::unique_ptr<G4VUserPhysicsList> _physicsList,
+                           std::string _physicsListName)
+    : logLevel(_logLevel),
+      runManager(std::move(_runManager)),
+      physicsList(_physicsList.release()),
+      physicsListName(std::move(_physicsListName)) {
+  if (runManager == nullptr) {
+    std::invalid_argument("runManager cannot be null");
+  }
+  if (physicsList == nullptr) {
+    std::invalid_argument("physicsList cannot be null");
+  }
+
+  // Set physics list
+  runManager->SetUserInitialization(physicsList);
+}
+
+Geant4Handle::~Geant4Handle() = default;
+
+void Geant4Handle::tweekLogging(int level) const {
+  Geant4Manager::tweekLogging(*runManager, level);
+}
+
+Geant4Manager& Geant4Manager::instance() {
+  static Geant4Manager manager;
+  return manager;
+}
+
+void Geant4Manager::tweekLogging(G4RunManager& runManager, int level) {
+  runManager.SetVerboseLevel(level);
+  G4EventManager::GetEventManager()->SetVerboseLevel(level);
+  G4EventManager::GetEventManager()->GetTrackingManager()->SetVerboseLevel(
+      level);
+  G4EventManager::GetEventManager()->GetStackManager()->SetVerboseLevel(level);
+
+  // Suppress the printing of physics information.
+#if G4VERSION_NUMBER >= 1100
+  G4HadronicParameters::Instance()->SetVerboseLevel(0);
+  G4HadronicProcessStore::Instance()->SetVerbose(0);
+  G4EmParameters::Instance()->SetIsPrintedFlag(true);
+#endif
+}
+
+std::shared_ptr<Geant4Handle> Geant4Manager::create(int logLevel,
```

Comment:
maybe rename to `createHandle` to make super clear that this is not a factory of `Geant4Manager`.